feature/settings/src/main/kotlin/de/tum/informatics/www1/artemis/native_app/feature/settings/SettingsScreen.kt

```diff
@@ -415,59 +418,149 @@ private fun BuildInformationSection(
 
-
 @Composable
-private fun PreferenceSection(
-    modifier: Modifier,
-    title: String,
-    entries: @Composable ColumnScope.() -> Unit
+fun PreferenceEntry(
+    modifier: Modifier = Modifier,
+    text: String,
+    icon: ImageVector? = null,
+    valueText: String? = null,
+    onClick: () -> Unit,
 ) {
-    Column(modifier = modifier) {
-        val childModifier = Modifier.fillMaxWidth()
+    Box(
+        modifier = modifier
+            .clickable(onClick = onClick)
+    ) {
+        Row(
+            modifier = modifier
+                .padding(horizontal = 16.dp, vertical = 10.dp)
+        ) {
+            Row(
+                horizontalArrangement = Arrangement.spacedBy(10.dp)
+            ) {
+                icon?.let {
+                    Icon(
+                        imageVector = it,
+                        contentDescription = null
+                    )
+                }
 
-        PreferenceSectionTitle(
-            modifier = childModifier.padding(horizontal = 16.dp),
-            text = title
-        )
+                Text(
+                    text = text,
+                    style = MaterialTheme.typography.bodyLarge
+                )
+            }
+
+            Spacer(modifier = Modifier.weight(1f))
+
+            valueText?.let {
+                Text(
+                    text = it,
+                    style = MaterialTheme.typography.bodyMedium.copy(fontWeight = FontWeight.Bold)
+                )
 
-        entries()
+            }
+        }
     }
 }
 
 @Composable
-private fun PreferenceSectionTitle(modifier: Modifier, text: String) {
-    Box(modifier = modifier) {
-        Text(
-            modifier = Modifier
-                .fillMaxWidth()
-                .padding(vertical = 16.dp),
-            text = text,
-            style = MaterialTheme.typography.titleLarge
-        )
+fun ServerURLEntry(
+    modifier: Modifier = Modifier,
+    text: String,
+    valueText: String,
+    onClick: () -> Unit,
+) {
+    Box(
+        modifier = modifier
+            .clickable(onClick = onClick)
+    ) {
+        Column(
+            modifier = modifier
+                .padding(horizontal = 16.dp, vertical = 10.dp)
+        ) {
+            Text(
+                text = text,
+                style = MaterialTheme.typography.bodyLarge
+            )
+
+            Text(
+                text = valueText,
+                style = MaterialTheme.typography.bodyMedium.copy(fontWeight = FontWeight.Bold),
+                modifier = Modifier.fillMaxWidth()
+            )
+        }
     }
 }
 
 @Composable
-private fun PreferenceEntry(modifier: Modifier, text: String, onClick: () -> Unit) {
-    Box(modifier = modifier.clickable(onClick = onClick)) {
-        Row(modifier = Modifier.padding(16.dp)) {
+fun ButtonEntry(
+    modifier: Modifier = Modifier,
+    text: String,
+    isFocused: Boolean = false,
+    textColor: Color = MaterialTheme.colorScheme.onSurface,
+    onClick: () -> Unit,
+) {
+    Box(
```

Comment:
Would it be sufficient here to remove the box and just have the row as the row layout? This also applies to the other Entries

Reply:
Yes, you're right. :+1: